pkg/envoy/registry/types.go

```diff
@@ -14,9 +15,11 @@ var log = logger.New("proxy-registry")
 type ProxyRegistry struct {
 	ProxyServiceMapper
 
-	connectedProxies sync.Map
+	connectedProxies map[string]*envoy.Proxy
```

Comment:
nit: place the connected proxies field under the mutex field, which signifies what the mutex locks.